fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -0,0 +1,274 @@
+            scanRangeLocations.addToLocations(scanRangeLocation);
+
+            result.add(scanRangeLocations);
+        }
+    }
+
+    @Override
+    protected String debugString() {
+        MoreObjects.ToStringHelper helper = MoreObjects.toStringHelper(this);
+        helper.addValue(super.debugString());
+        helper.addValue("icebergTable=" + srIcebergTable.getName());
+        return helper.toString();
+    }
+
+    @Override
+    protected String getNodeExplainString(String prefix, TExplainLevel detailLevel) {
+        StringBuilder output = new StringBuilder();
+
+        output.append(prefix).append("TABLE: ").append(srIcebergTable.getName()).append("\n");
+
+        if (null != sortColumn) {
+            output.append(prefix).append("SORT COLUMN: ").append(sortColumn).append("\n");
+        }
+        if (!conjuncts.isEmpty()) {
+            output.append(prefix).append("PREDICATES: ").append(
+                    getExplainString(conjuncts)).append("\n");
+        }
+
+        output.append(prefix).append(String.format("cardinality=%s", cardinality));
+        output.append("\n");
+
+        output.append(prefix).append(String.format("avgRowSize=%s", avgRowSize));
+        output.append("\n");
+
+        output.append(prefix).append(String.format("numNodes=%s", numNodes));
+        output.append("\n");
+
+        return output.toString();
+    }
+
+    @Override
+    protected String getNodeVerboseExplain(String prefix) {
+        StringBuilder output = new StringBuilder();
+
+        output.append(prefix).append("TABLE: ").append(srIcebergTable.getName()).append("\n");
+
+        if (null != sortColumn) {
+            output.append(prefix).append("SORT COLUMN: ").append(sortColumn).append("\n");
+        }
+        if (!conjuncts.isEmpty()) {
+            output.append(prefix).append("PREDICATES: ").append(
+                    getExplainString(conjuncts)).append("\n");
+        }
+
+        output.append(prefix).append(String.format("avgRowSize=%s", avgRowSize));
+        output.append("\n");
+
+        output.append(prefix).append(String.format("numNodes=%s", numNodes));
+        output.append("\n");
+
+        return output.toString();
+    }
+
+    @Override
+    public int getNumInstances() {
+        return result.size();
+    }
+
+    @Override
+    public void computeStats(Analyzer analyzer) {
+        Optional<Snapshot> snapshot = IcebergUtil.getCurrentTableSnapshot(
+                srIcebergTable.getIcebergTable(), true);
+        if (snapshot.isPresent()) {
+            for (FileScanTask task : IcebergUtil.getTableScan(
+                    srIcebergTable.getIcebergTable(), snapshot.get(),
+                    icebergPredicates, true).planFiles()) {
+                totalBytes += task.length();
+                DataFile file = task.file();
+                cardinality += file.recordCount() * (task.length() / file.fileSizeInBytes());
```

Comment:
The div between `task.length()` and `file.fileSizeInBytes()` will always be 0 because it's a long-long division and the task.length() will always be less than file.fileSizeInBytes().  I will suggest to cast the task.length() to a double type.

Reply:
The statistics here are now useless, I will delete these codes and submit another pr.